src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseSimulationResource.java

```diff
@@ -0,0 +1,190 @@
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.CourseService;
+import de.tum.in.www1.artemis.service.ProgrammingExerciseSimulationService;
+import de.tum.in.www1.artemis.service.UserService;
+import de.tum.in.www1.artemis.service.feature.Feature;
+import de.tum.in.www1.artemis.service.feature.FeatureToggle;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+
+@RestController
+@RequestMapping(ProgrammingExerciseSimulationResource.Endpoints.ROOT)
+/**
+ * Only for local development
+ * Simulates the creation of a programming exercise without a local setup
+ */
+public class ProgrammingExerciseSimulationResource {
+
+    private final Logger log = LoggerFactory.getLogger(ProgrammingExerciseResource.class);
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private static final String ENTITY_NAME = "programmingExercise";
+
+    private final CourseService courseService;
+
+    private final UserService userService;
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    private final String packageNameRegex = "^[a-z][a-z0-9_]*(\\.[a-z0-9_]+)+[0-9a-z_]$";
+
+    private final Pattern packageNamePattern = Pattern.compile(packageNameRegex);
+
+    private final ProgrammingExerciseSimulationService programmingExerciseSimulationService;
+
+    public ProgrammingExerciseSimulationResource(CourseService courseService, UserService userService, AuthorizationCheckService authCheckService,
+            ProgrammingExerciseRepository programmingExerciseRepository, ProgrammingExerciseSimulationService programmingExerciseSimulationService) {
+        this.courseService = courseService;
+        this.userService = userService;
+        this.authCheckService = authCheckService;
+        this.programmingExerciseRepository = programmingExerciseRepository;
+        this.programmingExerciseSimulationService = programmingExerciseSimulationService;
+    }
+
+    /**
+     * POST /programming-exercises/no-local-setup: Setup a new programmingExercise
+     * This method creates a new exercise
+     * This exercise is only a SIMULATION for the testing of programming exercises without local setup
+     * @param programmingExercise the input to create/setup new exercise
+     * @return a Response Entity
+     */
+    @PostMapping(ProgrammingExerciseSimulationResource.Endpoints.NO_LOCAL_SETUP)
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    @FeatureToggle(Feature.PROGRAMMING_EXERCISES)
+    public ResponseEntity<ProgrammingExercise> setupProgrammingExerciseWithoutLocalSetup(@RequestBody ProgrammingExercise programmingExercise) {
+        log.debug("REST request to setup ProgrammingExercise : {}", programmingExercise);
+        if (programmingExercise.getId() != null) {
```

Comment:
I think some of the checks below are not necessary for a local setup, which would greatly reduce the code complexity here, e.g. title and short name checks, all programming language and package related checks, even the max score can be left out

Reply:
I removed the input checks and the connected test cases